### PR TITLE
reversed variant for --map-passwd and --map-group

### PR DESCRIPTION
**Pull request by sshilovsky**
--map-passwd-rev and --map-group-rev

My use case for --map-group-rev is mounting /dev into chroot tree with different UID/GID. The other one was added just for completeness

### CODE REVIEW

File: src/bindfs.c [hunk 1]
```diff
@@ -2366,6 +2370,8 @@ int main(int argc, char *argv[])
         OPT_OFFSET2("--map=%s", "map=%s", map, -1),
         OPT_OFFSET2("--map-passwd=%s", "map-passwd=%s", map_passwd, -1),
         OPT_OFFSET2("--map-group=%s", "map-group=%s", map_group, -1),
+        OPT_OFFSET2("--map-passwd-rev=%s", "map-passwd=%s", map_passwd_rev, -1),
+        OPT_OFFSET2("--map-group-rev=%s", "map-group=%s", map_group_rev, -1),
```
Comment (mpartel) [inline src/bindfs.c:2374]:
```suggestion
        OPT_OFFSET2("--map-passwd-rev=%s", "map-passwd-rev=%s", map_passwd_rev, -1),
        OPT_OFFSET2("--map-group-rev=%s", "map-group-rev=%s", map_group_rev, -1),
```

File: src/bindfs.c [hunk 2]
```diff
@@ -1695,6 +1695,8 @@ static void print_usage(const char *progname)
            " --map=user1/user2:...      Let user2 see files of user1 as his own.\n"
            " --map-passwd=<passwdfile>  Load uid mapping from <passwdfile>.\n"
            " --map-group=<groupfile>    Load gid mapping from <groupfile>.\n"
+           " --map-passwd-rev=<passwdfile>  Load reversed uid mapping from <passwdfile>.\n"
+           " --map-group-rev=<groupfile>    Load reversed gid mapping from <groupfile>.\n"
```
Comment (mpartel) [inline src/bindfs.c:1699]: Let's line these up :)
```suggestion
           " --map-passwd=...           Load uid mapping from passwd-like file.\n"
           " --map-group=...            Load gid mapping from group-like file.\n"
           " --map-passwd-rev=...       Load reversed uid mapping from  passwd-like file.\n"
           " --map-group-rev=...        Load reversed gid mapping from group-like file.\n"
```

**Review by mpartel — CHANGES_REQUESTED** [review]
Nice!
Could you also add a test case to `tests/test_bindfs.rb`? You can copy the `root_testenv("--map-passwd=...` block as a starting point.

**Comment by mpartel** [comment]
Released as 1.16.1.
Thanks!

(I left out the change for skipping the test if unshare failed, because I'm uncomfortable with silently skipping tests if something goes wrong. It'd be better to change the test system so that a single failure doesn't block other tests from running, but I have too little time to work on bindfs these days.)

**Comment by sshilovsky** [comment]
Thank you for prompt merge!

On `unshare`, it is a step in the process of initialization environment for that particular test. I think that it failing can only mean that the initialization is impossible here, so the test can be skipped, just like it's skipped currently based on parsing `man unshare`.

Although there is another possibility that my chroot, which I was running the tests inside, was set up wrong. I should look into that anyway.

**Comment by mpartel** [comment]
Yeah, making the tests work in a variety of environments is a totally valid way to look at it (and some test frameworks explicitly support it), I just tend to put a lot more weight on dangers of invisible failures.

**Pull request merged**